### DISCUSSION TITLE
.NET 6 `DateOnly` and `TimeOnly`

### DISCUSSION THREAD

**Discussion by myokeeh** [Azure/azure-mobile-apps#220] [Q&A] [answered]
Can we make use of `DateOnly` and `TimeOnly` types with the latest update to this service? If so, how does that map over to the Azure SQL Service table column types? How about the offline sync SQLite database?

**Comment by adrianhall** [comment, marked as answer]
I have not added support for DateOnly and TimeOnly as yet.  The current release will be compatible with .NET 5 and 6, so it requires only to use cross-compatible types.

**Reply by fileman** [reply]
Hi, when they could be added?
.net 5 support expired, final release is planned for .net 6 only, something else?

**Reply by adrianhall** [reply]
.NET 6 is not released as GA as yet, so we are still supporting .NET 5.

@myokeeh - you should open an issue to ask for support of DateOnly and TimeOnly.

**Comment by adrianhall** [comment]
I'm resurrecting this thread because I've started working on it.

There are several pieces that are needed before I can declare "done"

Server side:
1. Support for DateOnly and TimeOnly serialization and deserialization.
2. Support for Edm.Date and Edm.TimeOfDay in the OData methods.
3. A set of query capabilities in OData that allow ge/gt/le/lt/eq and month(), day(), year(), hour(), minute(), second()

Client side (online):
1. Support for DateOnly and TimeOnly in .NET Standard 2.0.
2. Support for DateOnly and TimeOnly serialization and deserialization.
3. Conversion of DateOnly and TimeOnly comparisons in the LINQ to OData converter.
4. Conversion of date.{Month,Year,Day} and time.{Hour,Minute,Second} in the LINQ to OData converter.

Client side (offline - sqlite):
1. Storage of the DateOnly and TimeOnly types in SQLite.
2. Conversion of DateOnly and TimeOnly for comparisons in the OData to SQL converter.
3. Conversion of the date.{Month,Year,Day} and time.{Hour,minute,second} in the OData to SQL converter.

Obviously, offline - litedb isn't a consideration yet because LiteDb isn't available client-side yet, but it should make the comparisons easier (since LiteDb understands dates and times much better than SQLite)

As of today, the server side is complete.  Client side is pending.
